### DISCUSSION TITLE
How do you do your character controller?

### DISCUSSION THREAD

**Discussion by PebbleCat-Studios** [mozilla/hubs#6467] [General Questions & Discussion] [answered]
I'm working on my A-Frame scene and I was going around to see if there was a scene with one so I could base mine on that, and from what I have seen, you are the only thing I could find. Players could move around with WASD and stick onto the ground... How do you do that?

**Comment by hrithikwins** [comment, marked as answer]
Hey @funing36 what you are refring to is known as navmesh
Hubs uses navmesh to limit users movement in a certain space
we use **spoke** to generate floorplan and blender for navmesh ... both are ideally the same and you can use either of them
If you are looking for building something on hubs the best thing you can do is join hubs discord

**Reply by PebbleCat-Studios** [reply]
Ok, thanks. I did some research and actually adopted the same method.
